### PR TITLE
Define test data for user and project, using new core package

### PR DESCRIPTION
**Pull request by emmenko**
Trying to play around and test the new `@commercetools-test-data/core` package for building test data models.

I have some questions regarding defining the generator/transformer for the user + project, but I need to have a chat with @tdeekens first to clarify some points.

### CODE REVIEW

**Review by tdeekens — COMMENTED** [review]
Looks razor sharp. Thanks for pushing the idea here.

**Comment by emmenko** [comment]
This now should correctly implement the transformers for the permissions -> allAppliedPermissions (etc).

As a result of this, there is a pending fix in the test-data repo: https://github.com/commercetools/test-data/pull/4

File: test-data/project/fields/applied-menu-visibilities/transformers.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const upperFirst = (value: string): string =>
+  value.charAt(0).toUpperCase() + value.slice(1);
+
+const transformers = {
+  graphql: Transformer<TMenuVisibilities, TAppliedMenuVisibilitiesGraphql>(
+    'graphql',
+    {
+      replaceFields: ({ fields }) =>
+        Object.entries(fields).map(([key, value]) => ({
+          __typename: 'AppliedMenuVisibilities',
```
Comment (tdeekens) [inline test-data/project/fields/applied-menu-visibilities/transformers.ts:17]: Can‘t this be an add field.

File: test-data/project/generator.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    // The following fields are built in the builder
+    owner: null,
+    suspension: null,
+    expiry: null,
+    permissions: null,
+    actionRights: null,
+    menuVisibilities: null,
+    dataFences: null,
+  },
+  postBuild: (project) => {
```
Comment (tdeekens) [inline test-data/project/generator.ts:28]: Might be possible setting it to null and then replaceField. To not depend in test-data-bot for it.

**Review by tdeekens — COMMENTED** [review]
Looks very very promising. Thanks for pushing the topic!

File: test-data/project/fields/applied-menu-visibilities/transformers.ts (hunk 1 above)
Comment (emmenko) [reply test-data/project/fields/applied-menu-visibilities/transformers.ts:17]: If you use `replaceFields` you can't (and does not make sense) to use `addFields` or `removeFields`.

**Comment by emmenko** [comment]
Anything else to add/change here? @tdeekens @adnasa

**Review by tdeekens — APPROVED** [review]
Nope. This is awesome.

**Review by adnasa — APPROVED** [review]
love it

**Pull request merged**